### ISSUE TITLE
Register overrides other container factories with same protocol signature

### ISSUE THREAD

**Issue by thomasflad** [hmlongco/Factory#7]
Hi Michael, 

thanks for this great package! I've found an issue and maybe you can lead me to the right direction. 

I have a SharedContainer similar to this:

```Swift
class MyModule: SharedContainer {
    static let approve = Factory<MyProtocol<Input, Output>> { wrap(Approve()) } 
    static let complete = Factory<MyProtocol<Input, Output>> { wrap(Complete()) }
    static let reset = Factory<MyProtocol<Input, Output>> { wrap(Reset()) }
}
```

And my ViewModel is injecting the dependencies:

```Swift
final class ViewModel: ObservableObject {
        @Injected(MyModule.approve) private var approve
        @Injected(MyModule.complete) private var complete
        @Injected(MyModule.reset) private var reset
        ...
}
```

My issue now is that Factory is overriding my mocks in tests with the latest definition for the same protocol signature. My assumption was that this handled per container property. (Like https://github.com/hmlongco/Resolver/blob/master/Documentation/Names.md)

```Swift
extension MyModule {
    static func mockApprove(data: Output?) {
        approve.register { Fake<Input, Output>(data) }
    }

    static func mockComplete(data: Output?) {
        complete.register { Fake<Input, Output>(data) }
    }

   static func mockReset(data: Output?) {
        reset.register { Fake<Input, Output>(data) }
    }
```

The test:

```Swift
class MyTest: XCTestCase {

    override class func setUp() {
        super.setUp()
        MyModule.mockComplete()
        MyModule.mockApprove() <-- MyModule.complete behaves after evaluation like approve
        MyModule.mockReset() <-- MyModule.approve and MyModule.complete behave after evaluation like reset
    }
    ...
}
```

**Comment by hmlongco** [comment]
Since the definitions are static it's not really the container but the Factory that should be distinct. But yes, registering a type affected any other container of the same type. 

🤦 This prevented something as simple as...

```swift
extension SharedContainer {
    static let a = Factory<String> { "A" } 
    static let b = Factory<String> { "B" }
```
At any rate, I think it's fixed in 1.0.7.

**Comment by thomasflad** [comment]
Thanks a lot @hmlongco! Works like charm with 1.0.7 👍🏼
